File: write-api-server-kotlin/src/main/kotlin/com/github/j5ik2o/adceet/api/write/use/case/CreateThreadUseCase.kt
package com.github.j5ik2o.adceet.api.write.use.case

import com.github.j5ik2o.adceet.api.write.domain.AccountId
import com.github.j5ik2o.adceet.api.write.domain.ThreadId
import java.util.concurrent.CompletionStage

interface CreateThreadUseCase {
  fun execute(threadId: ThreadId, accountId: AccountId): CompletionStage<ThreadId>
}

File: write-api-server-scala/src/main/scala/com/github/j5ik2o/adceet/api/write/use/case/CreateThreadUseCase.scala
package com.github.j5ik2o.adceet.api.write.use.`case`

import com.github.j5ik2o.adceet.api.write.domain.{ AccountId, ThreadId }

import scala.concurrent.{ ExecutionContext, Future }

trait CreateThreadUseCase {
  def execute(threadId: ThreadId, accountId: AccountId)(implicit ec: ExecutionContext): Future[ThreadId]
}
